edx_event_bus_kafka/internal/consumer.py

```diff
@@ -132,6 +138,39 @@ def _shut_down(self):
         """
         self._shut_down_loop = True
 
+    def _triage_error(self, e, msg):
+        """
+        Handle error and determine if we still want to commit the offset
+
+        Returns:
+            True if we want to commit the message offset, False if not
+            Raises if the error is a Kafka fatal error
+        """
+        # case 1: DB error - reconnect and try again by rewinding the offset
+        if isinstance(e, OperationalError):
+            error_code, _ = e.args
+            if error_code == MYSQL_LOST_CONNECTION_ERROR:
+                message_location = TopicPartition(msg.topic(), msg.partition(), msg.offset())
+                db.connection.close()
```

Comment:
Before doing a detailed review, I'm wondering what happens if we call `db.connection.connect()` at the start of each loop? Does that work? Does it prevent the initial failure? 

Note: if we do this, we'd want to first check that there is a connection defined. We could have a consumer with no db. Also, unless we connect on all connections, we'd need to document that the same should be done in the consumer when using non-default connections, like the read-only db.

Reply:
What do you mean connect on all connections?

Reply:
You can ignore the all-connections note. It was based on something I had read, but can’t find, and it would be a more exceptional case anyway.

For now, I’m wondering if `connect()` are the start of each loop works more similarly to Django requests, and fixes the connection as-needed.

Reply:
Scouring the internet and django docs, I can't find any examples of manually calling connect(), which makes me pretty hesitant to use it. We'd also have to manually handle closing it (I think) which opens us up to future possible connection leaks.

Reply:
1. Does it seem to resolve the issue? If not, it isn't worth looking more deeply. If so, it is worth additional consideration.
2. Maybe we could call connect only when we think it is already connected, to refresh the connection as-needed.
3. Is it possible that we already aren't closing connections properly when exiting the loop? Is that something we should be doing? Or is Django treating this as one long "request", and when we exit, it is handling all of its usual clean-up activities?

Reply:
For 3, I don't know specifics, but I can't imagine Django doesn't have built-in tools for managing the connection for a management command. Most of our management commands talk to the DB so we would notice if those weren't getting closed properly.

Reply:
I know Django will autoclose connections based on a configured timeout

Reply:
I'm not too worried about 3, but I will point out that most management commands end, and ours runs indefinitely, so there could be django clean-up code that never gets run.